### DISCUSSION TITLE
Sudden growth of unique active users after SDK update

### DISCUSSION THREAD

**Discussion by MatejVancik** [getsentry/sentry-java#2512] [Q&A]
Hello there! I spoted an issue in our dashboards which I believe is relationated with changes in sentry SDK for android. We did an update of android SDK from `6.1.4` to `6.10.0` and gradle plugin from `3.0.1` to `3.4.0` at the beginning of the january and ever since we see ~3x times more `count_unique(user)` in dashboards than before. You can see that on the screenshot showing `count_unique(user)` grouped by app release in past 90 days.

<img width="950" alt="Screenshot 2023-02-02 at 10 15 05" src="https://user-images.githubusercontent.com/6854864/216299663-75a9144e-541e-4c85-bacc-4621d818706b.png">

I can confirm that these are not real numbers of active daily users as we don't see such growth in any of the others systems (PlayStore, etc). The reality is that the number of active users is pretty stable over past 90 days.

Another place where we see the effects of this is on the project details page on `Crash Free Sessions` and `Crash Free Users` which sharply went up after release with updated SDK. You can see it here:

<img width="1014" alt="Screenshot 2023-02-02 at 10 20 41" src="https://user-images.githubusercontent.com/6854864/216299717-05f4d227-190c-433e-a633-ba5ac5528edd.png">

We didn't do any other change to Sentry configuration in our app so the only obvious reason that I see would be some changes in updated SDK. Would anybody be able to confirm my theory or can think of any other reason why unique users would grow suddenly like that? Thanks!

**Comment by kahest** [comment]
Hello @MatejVancik! Thank you for reporting your findings, just wanted to let you know we are already looking into this.

**Reply by MatejVancik** [reply]
Thanks @kahest! Will keep an eye on the thread! 🙂

**Reply by romtsn** [reply]
Hi @MatejVancik, apologies for the late reply, this went completely out of my sight. Do you have more info on your app's behavior, e.g. does it use WorkManager or something alike that wakes the app in background? 

We've recently moved `session.start()` call from Application.onStart to when you call `SentryAndroid.init` [here](https://github.com/getsentry/sentry-java/pull/2356) - this might explain the growth in sessions.

**Reply by MatejVancik** [reply]
Thanks @romtsn I think that explains the issue! We call `SentryAndroid.init` in application's `onCreate` (like suggested in the [manual initialization](https://docs.sentry.io/platforms/android/configuration/manual-init/)) which can be called also when we receive a push notification (which we do a lot). 

So I think the obvious fix for us is to call `SentryOptions.setEnableAutoSessionTracking(false)` within `SentryAndroid.init` config and then add `Sentry.startSession()` in Application's `onStart`, correct? 

I understand the motivation behind the change you've introduced to keep the crashing session tracked correctly but as you can see with the issue we are experiencing, it's breaking the count of sessions and stability when app wakes up in the background.
If this is desired behaviour I think it would be worth updating the docs, because the [Release health - sessions](https://docs.sentry.io/platforms/android/configuration/releases/#sessions) says at the moment:
> A session represents the interaction between the user and the application.

Which is not completely true, considering a session is now counted also without any interaction from user.

**Reply by romtsn** [reply]
Yeap, no action from you required - we're intending to fix this by not starting the session for background wake-ups, so the behavior should remain unchanged in the fixed version. Will update here once the fixed version is released. Thanks for bearing with us!

**Reply by MatejVancik** [reply]
Perfect! Thanks @romtsn! Will wait for the update then!

**Reply by kahest** [reply]
[Version 6.14.0](https://github.com/getsentry/sentry-java/releases/tag/6.14.0) with a fix for this was just released. @MatejVancik thanks again for raising this!

**Reply by MatejVancik** [reply]
Thank you @kahest ! We're about to release an update with new version. Will let you know how it went 😉

**Reply by MatejVancik** [reply]
@kahest the fix definitely worked well! We're getting back to "normal" numbers. Thanks again!
<img width="1138" alt="Screenshot 2023-02-27 at 09 24 49" src="https://user-images.githubusercontent.com/6854864/221511832-3c74f3fc-0dfa-4f38-aafb-1f6026367991.png">

**Reply by romtsn** [reply]
@MatejVancik thanks for confirming, appreciated 👍

**Reply by romtsn** [reply]
@MatejVancik may I ask you to confirm if the crash-free sessions also went down?

**Reply by MatejVancik** [reply]
@romtsn Yes, crash-free sessions are also back to normal. Graph went down, now showing real numbers 😉
